zilliqa/src/api/zil.rs

```diff
@@ -175,6 +175,12 @@ fn create_transaction(
 
     // TODO: Perform some initial validation of the transaction
 
+    // If we don't trap this here, it will later cause the -1 in
+    // transaction::get_nonce() to panic.
+    if transaction.nonce == 0 {
```

Comment:
I wonder if we should make `nonce: NonZeroU64` in `TxZilliqa`?

Reply:
I think that's probably a good idea ... I can issue a PR if you like?

Reply:
Happy to do it separately, since this change is good in its own right :)